x-pack/platform/plugins/shared/streams_app/test/scout/ui/fixtures/page_objects/streams_app.ts

```diff
@@ -14,44 +14,183 @@ export class StreamsApp {
     await expect(this.page.getByText('StreamsTechnical Preview')).toBeVisible();
   }
 
-  async gotoStreamsFromBreadcrumb() {
-    await this.page
-      .getByTestId('breadcrumbs')
-      .getByRole('link', { name: 'Streams', exact: true })
-      .click();
+  async gotoStreamManagementTab(streamName: string, tabName: string) {
+    await this.page.gotoApp(`streams/${streamName}/management/${tabName}`);
+  }
+
+  async gotoPartitioningTab(streamName: string) {
+    await this.gotoStreamManagementTab(streamName, 'route');
+  }
+
+  async gotoDataRetentionTab(streamName: string) {
+    await this.gotoStreamManagementTab(streamName, 'lifecycle');
+  }
+
+  async gotoProcessingTab(streamName: string) {
+    await this.gotoStreamManagementTab(streamName, 'enrich');
+  }
+
+  async gotoSchemaEditorTab(streamName: string) {
+    await this.gotoStreamManagementTab(streamName, 'schemaEditor');
+  }
+
+  async gotoSignificantEventsTab(streamName: string) {
+    await this.gotoStreamManagementTab(streamName, 'significantEvents');
+  }
+
+  async gotoAdvancedTab(streamName: string) {
+    await this.gotoStreamManagementTab(streamName, 'advanced');
+  }
+
+  // Routing-specific utility methods
+  async clickCreateRoutingRule() {
+    await this.page.getByTestId('streamsAppStreamDetailRoutingAddRuleButton').click();
+  }
+
+  async fillRoutingRuleName(name: string) {
+    await this.page.getByTestId('streamsAppRoutingStreamEntryNameField').fill(name);
+  }
+
+  async clickEditRoutingRule(streamName: string) {
+    await this.page.getByTestId(`routingRuleEditButton-${streamName}`).click();
+  }
+
+  async saveRoutingRule() {
+    await this.page.getByRole('button', { name: 'Save' }).click();
+  }
+
+  async updateRoutingRule() {
+    await this.page.getByRole('button', { name: 'Change routing' }).click();
+  }
+
+  async cancelRoutingRule() {
+    await this.page.getByRole('button', { name: 'Cancel' }).click();
+  }
+
+  async removeRoutingRule() {
+    await this.page.getByRole('button', { name: 'Remove' }).click();
   }
 
-  async gotoStream(stream: string) {
-    const last = await this.page.getByTestId('breadcrumb last').textContent();
-    if (last !== 'Streams') {
-      await this.gotoStreamsFromBreadcrumb();
+  getDeleteModal() {
+    return this.page.getByRole('dialog');
+  }
+
+  async confirmDeleteInModal() {
+    await this.getDeleteModal().getByRole('button', { name: 'Delete' }).click();
```

Comment:
Remembering toasts issues in FTR tests, maybe we add a check that modal is gone?

Reply:
Addressed with [tests(streams): add check for closed modal](https://github.com/elastic/kibana/pull/229791/commits/83c046b02d80bd823e551248f9763a1a58b3249c)